conjure-undertow-processor/src/main/java/com/palantir/conjure/java/undertow/processor/data/EndpointDefinitions.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.processor.data;
+
+import com.google.auto.common.MoreElements;
+import com.google.common.base.Predicates;
+import com.palantir.conjure.java.undertow.annotations.Handle;
+import com.palantir.conjure.java.undertow.annotations.HttpMethod;
+import com.palantir.conjure.java.undertow.processor.ErrorContext;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+
+public final class EndpointDefinitions {
+
+    private final ParamTypesResolver paramTypesResolver;
+    private final HttpPathParser httpPathParser;
+    private final ArgumentTypesResolver argumentTypesResolver;
+    private final ReturnTypesResolver returnTypesResolver;
+
+    public EndpointDefinitions(ErrorContext errorContext, Elements elements, Types types) {
+        ResolverContext context = new ResolverContext(errorContext, elements, types);
+        this.paramTypesResolver = new ParamTypesResolver(context);
+        this.httpPathParser = new HttpPathParser(context);
+        this.argumentTypesResolver = new ArgumentTypesResolver(context);
+        this.returnTypesResolver = new ReturnTypesResolver(context);
+    }
+
+    public Optional<EndpointDefinition> tryParseEndpointDefinition(ExecutableElement element) {
+        AnnotationReflector requestAnnotationReflector = MoreElements.getAnnotationMirror(element, Handle.class)
+                .toJavaUtil()
+                .map(ImmutableAnnotationReflector::of)
+                .orElseThrow();
+
+        EndpointName endpointName =
+                ImmutableEndpointName.of(element.getSimpleName().toString());
+        ServiceName serviceName = ImmutableServiceName.of(
+                element.getEnclosingElement().getSimpleName().toString());
+
+        HttpMethod method = HttpMethod.valueOf(requestAnnotationReflector
```

Comment:
I wonder if we also need to gracefully handle parsing the http method, similar to the `maybeHttpPath` and `maybeReturnType` below?